client/homebrew/pages/editPage/editPage.jsx

```diff
@@ -233,6 +251,41 @@ const EditPage = createClass({
 		}));
 	},
 
+	updateHistory : function(){
+		const historyKeys = [];
+		[1,2,3,4,5].forEach((i)=>{
+			historyKeys.push(`${HISTORY_PREFIX}-${this.props.brew.shareId}-${i}`);
+		});
+		historyKeys.forEach((key, i)=>{
```

Comment:
I know this is a work in progress, but I have some suggestions on the logic sequence.

Pseudocode:
1) Check if any slot has expired:
  a) Check from oldest to newest (i.e. from slot 5 to 1) for expiry time vs current time
  b) If the slot is expired, remove it, and stop checking for expiry of any other slots (expire only one slot at a time)
2) Check if there is any empty slot and fill it:
  a) Check from oldest to newest (i.e., from slot 5 to 1) for an empty slot.
  b) If an empty slot is found:
    i)  If any newer slot above is already filled, move the next newest slot to fill the space
    ii) If If no newer slot above is already filled, create a new entry from the current brew
    iii) However the slot was filled, reset the expiry date for that entry based on its new slot position, even if it was already expired
  c) continue checking for the next empty slot and repeat (i.e. cascade all slots down as needed until no empty slot remains, resetting their expiry timers for any entry that was moved)
  
Thus, there will always be 1 entry at least 24 hours old (sitting in the 5-day slot), 1 entry 1 hour old (sitting in the 24 hour slot), etc, and we aren't expiring 80% of the entries at once. They all "graduate" to the next level when an older one dies; they don't all die together.

Reply:
I realized the logic was critically flawed last night - for example, a slot with a 5 day timer, edited continuously for 5 days and 5 minutes, would contain data that is 5 minutes old. That's clearly not the intended functionality!

Thanks for digging out the pseudocode, I was about to go look for it.

Reply:
This should also now be resolved with the latest commits. There's still some way to go on actual UI, but the underlying data is there and updating as expected.